btk/create_observing_generator.py

```diff
@@ -23,12 +23,14 @@ def __init__(
         if survey_name not in btk.survey.surveys:
             raise KeyError("Survey not implemented.")
 
-        if obs_conditions == None:
-            obs_conditions = DefaultObsConditions(survey_name, "i", stamp_size)
-        self.survey_name = survey_name
+        
         self.bands = btk.survey.surveys[survey_name]["bands"]
         self.stamp_size = stamp_size
-
+        if obs_conditions == None:
+            obs_conditions = {}
+            for band in self.bands:
+                obs_conditions[band] = DefaultObsConditions(survey_name, band, stamp_size)
```

Comment:
I think I understand your idea now @thuiop , looks good

Reply:
As it is written right now, the user has to provide the dictionary. I think it would be easier for them to give the custom class and have the dictionary and objects created internally. But that means it would not be fully consistent with the sampling function where we give the objet directly instead of the class. Maybe we should change it for the sampling function ?